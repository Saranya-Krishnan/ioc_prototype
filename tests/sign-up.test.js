
import React from 'react';
import SignUp from '../views/src/components/sign-up-form';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router'

test('it displays the sign up form', () =>{
    const component = renderer.create(
        <MemoryRouter>
            <SignUp></SignUp>
        </MemoryRouter>
    );
    let c = component.toJSON();
    expect(c).toMatchSnapshot();
});