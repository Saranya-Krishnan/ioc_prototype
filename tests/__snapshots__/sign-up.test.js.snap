// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays the sign up form 1`] = `
<div>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        * First Name
      </label>
      <input
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="First Name"
        value=""
      />
    </div>
    <div
      className="ui hidden negative message"
    >
      <p>
        First Name is Required
      </p>
    </div>
    <div
      className="field"
    >
      <label>
        Last Name
      </label>
      <input
        name="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label>
        * Email
      </label>
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
    </div>
    <div
      className="ui hidden negative message"
    >
      <p>
        Invalid email format
      </p>
    </div>
    <div
      className="field"
    >
      <label>
        * Password
      </label>
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label>
        * Confirm Password
      </label>
      <input
        name="password_confirm"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="ui hidden negative message"
    >
      <p>
        Passwords do not match.
      </p>
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          name="doAgree"
          readOnly={true}
          tabIndex={0}
          type="checkbox"
          value={undefined}
        />
        <label>
          * I agree to the Terms and Conditions
        </label>
      </div>
    </div>
    <button
      className="ui disabled button"
      onClick={[Function]}
      tabIndex={-1}
      type="submit"
    >
      Submit
    </button>
  </form>
  
</div>
`;
